### PR TITLE
Adding deauth code option

### PR DESCRIPTION
**Pull request by pocchoc**
In response to issue 97, providing ability to add deauth code

### CODE REVIEW

File: src/aireplay-ng.c [hunk 1]
```diff
@@ -1336,19 +1338,24 @@ int do_attack_deauth( void )
 
         if( memcmp( opt.r_dmac, NULL_MAC, 6 ) != 0 )
         {
-            /* deauthenticate the target */
+	    /* deauthenticate the target */
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: Could you not modify this indentation (even if incorrect)?

File: src/aireplay-ng.c [hunk 2]
```diff
@@ -1336,19 +1338,24 @@ int do_attack_deauth( void )
...
+            
+	    memcpy( h80211, DEAUTH_REQ, 26 );
+            
+	    memcpy( h80211 + 16, opt.r_bssid, 6 );
 
-            memcpy( h80211, DEAUTH_REQ, 26 );
-            memcpy( h80211 + 16, opt.r_bssid, 6 );
+            /* add the deauth code */
+            memcpy( h80211 + 24, &(opt.deauthreason), 1);
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: I don't think memcpy is necessary here, we're modifying a single byte of data.

File: src/aireplay-ng.c [hunk 3]
```diff
@@ -1418,13 +1426,16 @@ int do_attack_deauth( void )
...
-            PCT; printf( "Sending DeAuth to broadcast -- BSSID:"
+    	    PCT; printf( "Sending DeAuth (code %i) to broadcast -- BSSID:"
                          " [%02X:%02X:%02X:%02X:%02X:%02X]\n",
+                         opt.deauthreason,
                          opt.r_bssid[0], opt.r_bssid[1],
                          opt.r_bssid[2], opt.r_bssid[3],
                          opt.r_bssid[4], opt.r_bssid[5] );
 
             memcpy( h80211, DEAUTH_REQ, 26 );
+            memcpy( h80211 + 24, &(opt.deauthreason), 1);
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: Same as previous, I don't think memcpy is necessary, it's a single byte.

File: src/aireplay-ng.c [hunk 4]
```diff
@@ -238,6 +239,7 @@ struct options
     int f_fromds;
     int f_iswep;
 
+    int deauthreason;
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: You should use unsigned char instead.

File: src/aireplay-ng.c [hunk 5]
```diff
@@ -1418,13 +1426,16 @@ int do_attack_deauth( void )
...
+    	    PCT; printf( "Sending DeAuth (code %i) to broadcast -- BSSID:"
                          " [%02X:%02X:%02X:%02X:%02X:%02X]\n",
+                         opt.deauthreason,
                          opt.r_bssid[0], opt.r_bssid[1],
                          opt.r_bssid[2], opt.r_bssid[3],
                          opt.r_bssid[4], opt.r_bssid[5] );
 
             memcpy( h80211, DEAUTH_REQ, 26 );
+            memcpy( h80211 + 24, &(opt.deauthreason), 1);
+
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: Please remove the empty line

File: src/aireplay-ng.c [hunk 6]
```diff
@@ -6654,6 +6666,17 @@ int main( int argc, char *argv[] )
                 }
                 break;
 
+            case 'Z' :
+
+                ret = sscanf( optarg, "%d", &opt.deauthreason );
```
Comment (aircrack-ng) [inline src/aireplay-ng.c]: This might need to be modified when switching to unsigned char for opt.deauthreason

**Review by aircrack-ng — CHANGES_REQUESTED** [review]
Could you update the patch according to the review?

**Comment by pocchoc** [comment]
Thanks for the pointers, I've updated the sections as requested and re-tested.

**Comment by aircrack-ng** [comment]
Committed in [r2930](http://trac.aircrack-ng.org/changeset/2930) (fixed indentation of some code) and slightly modified in [r2931](http://trac.aircrack-ng.org/changeset/2931).

**Pull request closed without merging**